dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateFactoryImpl.java

```diff
@@ -124,20 +122,25 @@ public void save(Template template, String existingId) throws DotDataException {
 
 	public void deleteFromCache(Template template) throws DotDataException {
 		templateCache.remove(template.getInode());
-		//WorkingCache.removeAssetFromCache(template);
-		//if (template.isLive()) {
-		//	LiveCache.removeAssetFromCache(template);
-		//}
 		CacheLocator.getIdentifierCache().removeFromCacheByVersionable(template);
 	}
 
 	@SuppressWarnings("unchecked")
 	public Template findWorkingTemplateByName(String name, Host host) throws DotDataException {
-		HibernateUtil hu = new HibernateUtil(Template.class);
-		hu.setSQLQuery(templateWithNameSQL);
-		hu.setParam(host.getIdentifier());
-		hu.setParam(name);
-		return (Template) hu.load();
+		DotConnect dc = new DotConnect();
+		dc.setSQL(templateWithNameSQL);
+		dc.addParam(host.getIdentifier());
+		dc.addParam(name);
+		try{
+			List<Template> result = ConvertToPOJOUtil.convertDotConnectMapToTemplate(dc.loadResults());
+			if (result!= null && !result.isEmpty()){
+				return result.get(0);
+			}
+		}catch (Exception e){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)